File: oop/heritage/exercice7/src/Cylinder.java
public class Cylinder extends Circle {
    protected double hauteur;

    public Cylinder(double x, double y, double rayon, double hauteur) {
        super(x, y, rayon);
        this.hauteur = hauteur;
    }

    @Override
    public double surface() {
        return 2 * (3.14 * rayon * rayon) + 2 * 3.14 * rayon * hauteur;
    }

    @Override
    public void affiche() {
        System.out.println("Je suis un cylindre de surface " + surface());
    }
}

File: oop/heritage/CPP/exercice7/src/Cylinder.cpp
#include "Cylinder.hpp"

Cylinder::Cylinder(double x, double y, double rayon, double hauteur)
    : Circle(x, y, rayon), hauteur(hauteur) {}

double Cylinder::surface() const
{
    return 2 * (3.14 * rayon * rayon) + 2 * 3.14 * rayon * hauteur;
}

void Cylinder::affiche() const
{
    std::cout << "Je suis un cylindre de surface " << surface() << '\n';
}
